components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +384,182 @@ public void setAlias(String alias) {
+        }
+        List<CertificateInfo> certificateInfos = new ArrayList<>();
+        for (int i = 0; i < jsonCertificateInfoArray.length(); i++) {
+            JSONObject jsonCertificateInfoObject = (JSONObject) jsonCertificateInfoArray.get(i);
+
+            CertificateInfo certificateInfo = new CertificateInfo();
+            certificateInfo.setThumbPrint(IdentityApplicationManagementUtil.generateThumbPrint(
+                    jsonCertificateInfoObject.getString(THUMB_PRINT)));
+            certificateInfo.setCertValue(jsonCertificateInfoObject.getString(CERT_VALUE));
+            certificateInfos.add(certificateInfo);
+        }
+        return certificateInfos.toArray(new CertificateInfo[jsonCertificateInfoArray.length()]);
+    }
+
+    /**
+     * handle the certificate which is in encoded format.
+     *
+     * @param certificateValue
+     * @return array of certificate value and thumbPrint of each certificates.
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] handleEncodedCertificate(String certificateValue) throws NoSuchAlgorithmException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling encoded certificates.");
+        }
+        String decodedCertificate = new String(Base64.getDecoder().decode(certificateValue));
+        int numberOfCertificates = StringUtils.countMatches(decodedCertificate,
+                IdentityUtil.PEM_BEGIN_CERTFICATE);
+
+        if (numberOfCertificates > 1) {
+            if (log.isDebugEnabled()) {
+                log.debug("More than one encoded certificate has been found.");
+            }
+        }
+        List<CertificateInfo> certificateInfos = new ArrayList<>();
+        for (int i = 1; i <= numberOfCertificates; i++) {
+            String certificateVal;
+            if (i == numberOfCertificates) {
+                certificateVal = Base64.getEncoder().encodeToString(decodedCertificate.substring(StringUtils.
+                        ordinalIndexOf(decodedCertificate, IdentityUtil.PEM_BEGIN_CERTFICATE, i)).getBytes(StandardCharsets.UTF_8));
+            } else {
+                certificateVal = Base64.getEncoder().encodeToString(decodedCertificate.substring(StringUtils.ordinalIndexOf(
+                        decodedCertificate, IdentityUtil.PEM_BEGIN_CERTFICATE, i),
+                        StringUtils.ordinalIndexOf(
+                                decodedCertificate, IdentityUtil.PEM_BEGIN_CERTFICATE, i +
+                                        1)).getBytes(StandardCharsets.UTF_8));
+            }
+            CertificateInfo certificateInfo = new CertificateInfo();
+            certificateInfo.setThumbPrint(IdentityApplicationManagementUtil.generateThumbPrint(
+                    certificateVal));
+            certificateInfo.setCertValue(certificateVal);
+            certificateInfos.add(certificateInfo);
+        }
+        return certificateInfos.toArray(new CertificateInfo[numberOfCertificates]);
+    }
```

Comment:
Line #409 to #519 , Doesn't this duplicate some code from handlePlainTextCertificate method? try reusing code